File: DSA/Sarthak_Jain_590014936/DAY_11/Day_11_Question_3.py
#Sarthak Jain 590014936
#Day 11 Question 3
#Count Palindromic Substrings

s = input("Enter a string: ")

def count_palindromes_around_center(left, right):
    count = 0
    while left >= 0 and right < len(s) and s[left] == s[right]:
        count += 1
        left -= 1
        right += 1
    return count

total = 0

for i in range(len(s)):
    total += count_palindromes_around_center(i, i)     # odd-length palindromes
    total += count_palindromes_around_center(i, i + 1) # even-length palindromes

print("Total palindromic substrings:", total)

File: DSA/RATNA_KHANNA_590013141/DAY_11_QUESTION_3.c
#include <stdio.h>
#include <string.h>

int main() {
    char str[1000];
    printf("Enter a string: ");
    scanf("%s", str);

    int len = strlen(str);
    int count = 0;

    for (int center = 0; center < len; center++) {
        int left, right;

        left = center;
        right = center;
        while (left >= 0 && right < len && str[left] == str[right]) {
            count++;
            left--;
            right++;
        }
        left = center;
        right = center + 1;
        while (left >= 0 && right < len && str[left] == str[right]) {
            count++;
            left--;
            right++;
        }
    }

    printf("Total Palindromic Substrings: %d\n", count);

    return 0;
}
